labman/db/tests/integration_tests.py

```diff
@@ -0,0 +1,214 @@
+    shotgun_process = LibraryPrepShotgunProcess.create(
+        user, norm_plate, 'Test Shotgun Library %s' % datetime.now(), kappa,
+        stub, 4000, Plate(19), Plate(20))
+    shotgun_plate = shotgun_process.plates[0]
+    return shotgun_process, shotgun_plate
+
+
+def create_plate_pool_process(user, shotgun_quant_process, shotgun_plate,
+                              func_data):
+    input_compositions = []
+    echo = Equipment(8)
+    for well in chain.from_iterable(shotgun_plate.layout):
+        if well is not None:
+            input_compositions.append({
+                'composition': well.composition, 'input_volume': 1,
+                'percentage_of_output': 1/9.0})
+    pool_process = PoolingProcess.create(
+        user, shotgun_quant_process, 'New test pool name %s' % datetime.now(),
+        4, input_compositions, func_data, robot=echo)
+    return pool_process
+
+
+def create_pools_pool_process(user, quant_process, pools):
+    input_compositions = [
+        {'composition': p, 'input_volume': 1, 'percentage_of_output': 1/9.0}
+        for p in pools]
+    pool_process = PoolingProcess.create(
+        user, quant_process, 'New pool name %s' % datetime.now(), 5,
+        input_compositions, {"function": "amplicon_pool", "parameters": {}})
+    return pool_process
+
+
+def create_sequencing_process(user, pools):
+    seq_process = SequencingProcess.create(
+        user, pools, 'New sequencing run %s' % datetime.now(),
+        'Run experiment %s' % datetime.now(), Equipment(18), 151, 151,
+        User('admin@foo.bar'),
+        contacts=[User('test@foo.bar'), User('demo@microbio.me')])
+    return seq_process
+
+
+def amplicon_workflow(user, samples):
+    # Sample Plating
+    sp_process, sample_plate = create_sample_plate_process(user, samples[:96])
+    # gDNA extraction
+    ext_process, gdna_plate = create_gdna_extraction_process(
+        user, sample_plate)
+    # Amplicon library prep
+    amplicon_process, amplicon_plate = create_amplicon_prep(user, gdna_plate)
+    # Library plate quantification
+    amplicon_quant_process = create_quantification_process(
+        user, amplicon_plate)
+    # Plate pooling process
+    plate_pool_process = create_plate_pool_process(
+        user, amplicon_quant_process, amplicon_plate,
+        {'function': 'amplicon',
+         'parameters': {"dna_amount": 240, "min_val": 1, "max_val": 15,
+                        "blank_volume": 2, "robot": 6, "destination": 1}})
+    # Quantify pools
+    pool_quant_process = create_pool_quantification_process(
+        user, [plate_pool_process.pool])
+    # Create sequencing pool process
+    seq_pool_process = create_pools_pool_process(
+        user, pool_quant_process, [plate_pool_process.pool])
+    # Sequencing process
+    create_sequencing_process(user, [seq_pool_process.pool])
+
+
+def shotgun_workflow(user, samples):
+    # Sample Plating
+    sp_process, sample_plate = create_sample_plate_process(user, samples[:96])
```

Comment:
Typically these gDNA runs have four plates... But maybe not worth doing that for this integration test, since it's just exercising each of the steps rather than trying to actually load the system?

Reply:
That's correct - although I do want to extend the functionality of this functions to be able to handle a parameter with the number of plates to create.